src/main/java/com/coveo/pushapiclient/PlatformClient.java

```diff
@@ -444,6 +444,29 @@ public HttpResponse<String> pushFileContainerContent(String sourceId, FileContai
     return this.api.put(uri, headers, HttpRequest.BodyPublishers.ofString(""));
   }
 
+  /**
+   * Push a file container into a stream source. See [Push the File Container into a Stream
+   * Source](https://docs.coveo.com/en/l62e0540/coveo-for-commerce/how-to-update-your-catalog#step-3-send-the-file-container-to-update-your-catalog).
+   *
+   * @param sourceId
+   * @param fileContainer
+   * @return
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public HttpResponse<String> pushFileContainerContentToStreamSource(
+      String sourceId, FileContainer fileContainer) throws IOException, InterruptedException {
+    String[] headers =
+        this.getHeaders(this.getAuthorizationHeader(), this.getContentTypeApplicationJSONHeader());
+    URI uri =
+        URI.create(
+            this.getBasePushURL()
+                + String.format(
+                    "/sources/%s/stream/update?fileId=%s", sourceId, fileContainer.fileId));
+
+    return this.api.put(uri, headers, HttpRequest.BodyPublishers.ofString(""));
```

Comment:
Shouldn't the file container's content be part of the put ?

Reply:
No, that part is done in the previous step ([Upload document content into the file container](https://docs.coveo.com/en/l62e0540/coveo-for-commerce/how-to-update-your-catalog#step-2-upload-the-full-document-content-into-the-file-container))